platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -0,0 +1,305 @@
+
+    protected static final Logger LOGGER = LoggerFactory.getLogger(HttpSolrClientFactory.class);
+
+    private static ScheduledExecutorService executor = getThreadPool();
+
+    public static final String DEFAULT_CORE_NAME = "core1";
+
+    public static final List<String> DEFAULT_PROTOCOLS = Collections.unmodifiableList(
+            Arrays.asList(StringUtils.split(System.getProperty("https.protocols"), ",")));
+
+    public static final List<String> DEFAULT_CIPHER_SUITES = Collections.unmodifiableList(
+            Arrays.asList(StringUtils.split(System.getProperty("https.cipherSuites"), ",")));
+
+    public static final String DEFAULT_SCHEMA_XML = "schema.xml";
+
+    public static final String DEFAULT_SOLRCONFIG_XML = "solrconfig.xml";
+
+    private static final String THREAD_POOL_DEFAULT_SIZE = "128";
+
+    private static ScheduledExecutorService getThreadPool() throws NumberFormatException {
+        Integer threadPoolSize = Integer.parseInt(
+                System.getProperty("org.codice.ddf.system.threadPoolSize",
+                        THREAD_POOL_DEFAULT_SIZE));
+        return Executors.newScheduledThreadPool(threadPoolSize);
+    }
+
+    public static String getDefaultHttpsAddress() {
+        return SystemBaseUrl.constructUrl("https", "/solr");
+    }
+
+    public static String getDefaultHttpAddress() {
+        return SystemBaseUrl.constructUrl("http", "/solr");
+    }
+
+    /**
+     * Creates an {@link org.apache.solr.client.solrj.SolrClient} with the default http address
+     * url.
+     *
+     * @return SolrClient
+     */
+    static SolrClient getHttpSolrClient() {
+        return new HttpSolrClient(getDefaultHttpAddress());
+    }
+
+    @Override
+    public Future<SolrClient> newClient(String core) {
+        ConfigurationStore configStore = ConfigurationStore.getInstance();
+        if (System.getProperty("solr.data.dir") != null) {
+            configStore.setDataDirectoryPath(System.getProperty("solr.data.dir"));
+        }
+        String solrUrl = System.getProperty("solr.http.url", getDefaultHttpsAddress());
+        return getHttpSolrClient(solrUrl, core);
+    }
+
+    public static Future<SolrClient> getHttpSolrClient(String url) {
+        return getHttpSolrClient(url, DEFAULT_CORE_NAME, null);
+    }
+
+    public static Future<SolrClient> getHttpSolrClient(String url, String coreName) {
+        return getHttpSolrClient(url, coreName, null);
+    }
+
+    public static Future<SolrClient> getHttpSolrClient(String url, String coreName,
+            String configFile) {
+        String solrUrl = StringUtils.defaultIfBlank(url, SystemBaseUrl.constructUrl("/solr"));
+        String coreUrl = url + "/" + coreName;
+
+        RetryPolicy retryPolicy = new RetryPolicy()
+                .withBackoff(10, TimeUnit.MINUTES.toMillis(1), TimeUnit.MILLISECONDS);
+        return Failsafe.with(retryPolicy)
+                .with(executor)
+                .onRetry((c, failure, ctx) -> LOGGER.debug(
```

Comment:
I think it would be a `SolrClient` if the retry policy had a `retryWhen` or `retryIf` and the `get` returned a valid `SolrClient`.